airflow/models/asset.py

```diff
@@ -96,6 +98,30 @@ def resolve_ref_to_asset(
     return session.scalar(stmt)
 
 
+def resolve_assets_as_dag_dependencies(
+    *,
+    assets: list[Asset],
+    source: str,
+    target: str,
+    session: Session,
+) -> list[DagDependency]:
+    asset_id_names = session.scalars(
+        select(AssetModel.id, AssetAliasModel.name).where(
+            tuple_(AssetModel.name, AssetModel.uri).in_((asset.name, asset.uri) for asset in assets)
+        )
+    ).all()
+    return [
+        DagDependency(
+            source=source,
+            target=target,
+            label=asset_name,
+            dependency_type="asset",
+            dependency_id=str(asset_id),
+        )
+        for (asset_id, asset_name) in asset_id_names
+    ]
```

Comment:
```suggestion
    asset_id_names = session.execute(
        select(AssetModel.id, AssetAliasModel.name).where(
            tuple_(AssetModel.name, AssetModel.uri).in_((asset.name, asset.uri) for asset in assets)
        )
    )
    return [
        DagDependency(
            source=source,
            target=target,
            label=asset_name,
            dependency_type="asset",
            dependency_id=str(asset_id),
        )
        for (asset_id, asset_name) in asset_id_names
    ]
```

I don’t think using `scalars` here is correct since it only returns the first column?

Reply:
It used to be a case where we need to get all related assets. But I'll need to check again during rework

Reply:
This is removed